### PR TITLE
Add TLS support [13508]

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: docs/rst/user_manual/wan_configuration.rst [hunk 1]
```diff
@@ -68,13 +68,86 @@ These are a list of tips to help choosing whether to use one or the other.
...
+        - Mandatory for TLS servers
+        - Path to the public certificate chain file.
+
+    *   - ``dh_params``
+        - Mandatory for TLS servers
+        - Path to the Diffie-Hellman parameters file.
+
+Below is an example on how to configure a WAN participant as a TLS server and client:
+
+.. code-block:: yaml
```
Comment (jparisu) [inline docs/rst/user_manual/wan_configuration.rst:114]: Move this to a subsection below in the `Examples` Section of this same page.

File: docs/rst/user_manual/wan_configuration.rst [hunk 2]
```diff
@@ -68,13 +68,86 @@ These are a list of tips to help choosing whether to use one or the other.
...
+
+|eddsrouter| also supports `TLS over TCP <https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/tcp/tls.html>`_,
+and its configuration can be set per participant for types Local Discovery Server and WAN. Following is a list of the
+accepted entries under the ``tls`` tag:
+
+.. list-table::
+    :header-rows: 1
+
+    *   - Tag
+        - Requiredness
```
Comment (jparisu) [inline docs/rst/user_manual/wan_configuration.rst:89]: This word seems not common (and is not included in sphinx dictionary). I will use other, maybe `Required`

File: src/cpp/configuration/DiscoveryServerParticipantConfiguration.cpp [hunk 3]
```diff
@@ -77,6 +77,45 @@ std::vector<DiscoveryServerConnectionAddress> DiscoveryServerParticipantConfigur
     return result;
 }
 
+std::map<std::string, std::string> DiscoveryServerParticipantConfiguration::tls_configuration() const
+{
+    std::map<std::string, std::string> result;
+
+    if (raw_configuration_[TLS_TAG])
+    {
+        if (!raw_configuration_[TLS_TAG].IsMap() && !raw_configuration_[TLS_TAG].IsNull())
```
Comment (jparisu) [inline src/cpp/configuration/DiscoveryServerParticipantConfiguration.cpp:86]: Probably TLS should not be empty ¿?

**Review by jparisu — CHANGES_REQUESTED** [review]

File: docs/rst/user_manual/wan_configuration.rst (hunk 1 above)
Comment (juanlofer-eprosima) [reply docs/rst/user_manual/wan_configuration.rst]: :+1:

File: docs/rst/user_manual/wan_configuration.rst (hunk 2 above)
Comment (juanlofer-eprosima) [reply docs/rst/user_manual/wan_configuration.rst:89]: Not very common, probably old-fashioned or nonexistent, but for the Computer Science domain should be ok I think.

File: src/cpp/configuration/DiscoveryServerParticipantConfiguration.cpp (hunk 3 above)
Comment (juanlofer-eprosima) [reply src/cpp/configuration/DiscoveryServerParticipantConfiguration.cpp:86]: Don't know, I thought it's not enough for breaking.

**Review by jparisu — APPROVED** [review]
LGTM
Waiting CI

**Pull request merged**